common/display/displayqueue.cpp

```diff
@@ -436,7 +438,16 @@ bool DisplayQueue::QueueUpdate(std::vector<HwcLayer*>& source_layers,
     RecyclePreviousPlaneSurfaces();
     render_layers = display_plane_manager_->ValidateLayers(
         layers, cursor_layers, state_ & kConfigurationChanged,
-        idle_frame || disable_ovelays, current_composition_planes);
+        idle_frame || disable_ovelays || (state_ & kApplyColorMatrix),
```

Comment:
We don't need this (state_ & kApplyColorMatrix) here as below we are forcing GPU Composition for different planes.

Reply:
Android launcher wallpaper could not be correctly rendered after remove this, seems there is something wrong in layer calculation.

Reply:
@gaochun  if some layer is getting messed up, let's try to understand which one. This should be fixed in hwc anyway.

Reply:
The wallpaper layer data is HAL_PIXEL_FORMAT_RGBX_8888 format, which will get wrong color after the ctm transform, I'm checking whether there is issue of processing alpha data.

Reply:
@kalyankondapally I took some time to enable gpu rendering of each plane while keeping overlay enabled, but there are still some issues and regressions to fix. As the color correction and inversion need to be enabled by  ww42 and I'm OOO util 10/19, can I  keep gpu compositing here and add a TODO tag first? I'll submit a new path to enable overlays after all issues get fixed.

Reply:
@gaochun This broke APL platforms completely.

Reply:
@kalyankondapally Verified with @js0701 's Broxton device, the patch works, did you add "BOARD_DISABLE_DRM_CTM := true"  to BoardConfig.mk?